### ISSUE TITLE
Failed to generate types for contract ABI: Conversion from 'ethereum' to 'AssemblyScript' for source type 'MyLibrary.MyEnum' is not supported. Code 1.

### ISSUE THREAD

**Issue by nicolasguasca1** [graphprotocol/graph-tooling#1423]
### Which packages are impacted by your issue?

@graphprotocol/graph-cli, @graphprotocol/graph-ts

### Describe the issue

After trying to 'init' a subgraph based on this contract on Sepolia testnet: 0x69Bb9445f6f12e0fC3194d1bac2da3105AB49596.

I get the above mentioned error. The enum in question is this one: 
```enum PercentApproved {
    none,
    quarter,
    half,
    threequarter,
    all
  }```
  
  To reproduce the error one can simply run 'graph init --studio' and follow the steps to deploy on Sepolia. With the latest graph-cli (0.55) I get graph-ts 0.30 instead of graph-ts 0.31. Does it have to do with that?
  
  Im out of clue here. I could move on writing the schema but Im affraid that the mappings are referencing a 'generated' folder that doesn't exist (as the 'graph init' command failed)
  
  

### Reproduction

You can reproduce the example by trying to 'init' the contract 0x69Bb9445f6f12e0fC3194d1bac2da3105AB49596 on Sepolia testnet.

### Steps to Reproduce the Bug or Issue

1. graph init --studio lomplay_ev
2. Follow the step to deploy on Sepolia
*Either setting index-events as entities or not, the result is the same.


### Expected behavior

The command should finish the execution having all generated files and not half of them. 

### Screenshots or Videos

_No response_

### Platform

- OS: macOS
- NodeJS: 18.16.0
- `@graphprotocol/*` version(s): 0.55.0

- `graph-ts/` version(s): 0.30.0 // Adding 0.31.0 has the same result


### Subgraph Manifest

specVersion: 0.0.5
schema:
  file: ./schema.graphql
dataSources:
  - kind: ethereum
    name: EventLibrary
    network: sepolia
    source:
      address: "0x69Bb9445f6f12e0fC3194d1bac2da3105AB49596"
      abi: EventLibrary
      startBlock: 3942013
    mapping:
      kind: ethereum/events
      apiVersion: 0.0.7
      language: wasm/assemblyscript
      entities:
        - EventAdded
        - EventDeleted
        - EventSeriesCreated
        - EventSeriesStateChange
        - EventUpdated
        - EventsList
        - LomEvTokensUpdate
        - MarketAllowed
        - MetadataHashChange
        - SaleExecuted
        - TierAdded
        - TierSales
        - TierUpdated
        - TiersDeleted
        - TiersList
        - TiersToMarket
      abis:
        - name: EventLibrary
          file: ./abis/EventLibrary.json
      eventHandlers:
        - event: EventAdded(indexed address,(uint256,(uint32,uint32,uint32,(uint256,uint256),address,bytes32,string,uint8,uint8,bool)))
          handler: handleEventAdded
        - event: EventDeleted(indexed address,(indexed uint256,(uint32,uint32,uint32,(uint256,uint256),address,bytes32,string,uint8,uint8,bool)))
          handler: handleEventDeleted
        - event: EventSeriesCreated(indexed address,indexed address,indexed string)
          handler: handleEventSeriesCreated
        - event: EventSeriesStateChange(indexed address,indexed uint8,(uint256,(uint32,uint32,uint32,(uint256,uint256),address,bytes32,string,uint8,uint8,bool))[])
          handler: handleEventSeriesStateChange
        - event: EventUpdated(indexed address,uint256,(uint256,(uint32,uint32,uint32,(uint256,uint256),address,bytes32,string,uint8,uint8,bool)))
          handler: handleEventUpdated
        - event: EventsList((uint256,(uint32,uint32,uint32,(uint256,uint256),address,bytes32,string,uint8,uint8,bool))[])
          handler: handleEventsList
        - event: LomEvTokensUpdate(indexed uint256[],indexed uint256[],indexed address)
          handler: handleLomEvTokensUpdate
        - event: MarketAllowed(indexed address,indexed uint256,indexed address)
          handler: handleMarketAllowed
        - event: MetadataHashChange(indexed uint256,string)
          handler: handleMetadataHashChange
        - event: SaleExecuted(indexed address,indexed address,indexed uint256[])
          handler: handleSaleExecuted
        - event: TierAdded(indexed uint256,indexed uint256,bytes32)
          handler: handleTierAdded
        - event: TierSales(indexed uint256,indexed uint256[],indexed uint256[])
          handler: handleTierSales
        - event: TierUpdated(indexed address,indexed uint256,(indexed uint256,(uint256,uint256,uint256,uint32,uint32,uint32,uint32,uint8,string,bool)))
          handler: handleTierUpdated
        - event: TiersDeleted(indexed address,(indexed uint256,(uint256,uint256,uint256,uint32,uint32,uint32,uint32,uint8,string,bool))[])
          handler: handleTiersDeleted
        - event: TiersList((indexed uint256,(uint256,uint256,uint256,uint32,uint32,uint32,uint32,uint8,string,bool))[])
          handler: handleTiersList
        - event: TiersToMarket(indexed address,(indexed uint256,(uint256,uint256,uint256,uint32,uint32,uint32,uint32,uint8,string,bool))[])
          handler: handleTiersToMarket
      file: ./src/event-library.ts


### Subgraph GraphQL Schema

type EventAdded @entity(immutable: true) {
  id: Bytes!
  modifierAccount: Bytes! # address
  eventAdded_index: BigInt! # uint256
  eventAdded_indexedEvent_ticketNumber: BigInt! # uint32
  eventAdded_indexedEvent_reservedCapacity: BigInt! # uint32
  eventAdded_indexedEvent_tierCount: BigInt! # uint32
  eventAdded_indexedEvent_dateRange_start: BigInt! # uint256
  eventAdded_indexedEvent_dateRange_end: BigInt! # uint256
  eventAdded_indexedEvent_organizer: Bytes! # address
  eventAdded_indexedEvent_name: Bytes! # bytes32
  eventAdded_indexedEvent_subEvURI: String! # string
  eventAdded_indexedEvent_eventState: Int! # uint8
  eventAdded_indexedEvent_eventType: Int! # uint8
  eventAdded_indexedEvent_exist: Boolean! # bool
  blockNumber: BigInt!
  blockTimestamp: BigInt!
  transactionHash: Bytes!
}

type EventDeleted @entity(immutable: true) {
  id: Bytes!
  modifierAccount: Bytes! # address
  eventDeleted_index: BigInt! # uint256
  eventDeleted_indexedEvent_ticketNumber: BigInt! # uint32
  eventDeleted_indexedEvent_reservedCapacity: BigInt! # uint32
  eventDeleted_indexedEvent_tierCount: BigInt! # uint32
  eventDeleted_indexedEvent_dateRange_start: BigInt! # uint256
  eventDeleted_indexedEvent_dateRange_end: BigInt! # uint256
  eventDeleted_indexedEvent_organizer: Bytes! # address
  eventDeleted_indexedEvent_name: Bytes! # bytes32
  eventDeleted_indexedEvent_subEvURI: String! # string
  eventDeleted_indexedEvent_eventState: Int! # uint8
  eventDeleted_indexedEvent_eventType: Int! # uint8
  eventDeleted_indexedEvent_exist: Boolean! # bool
  blockNumber: BigInt!
  blockTimestamp: BigInt!
  transactionHash: Bytes!
}

type EventSeriesCreated @entity(immutable: true) {
  id: Bytes!
  organizer: Bytes! # address
  callerAccount: Bytes! # address
  changeableURI: String! # string
  blockNumber: BigInt!
  blockTimestamp: BigInt!
  transactionHash: Bytes!
}

type EventSeriesStateChange @entity(immutable: true) {
  id: Bytes!
  modifierAccount: Bytes! # address
  state: Int! # uint8
  eventSeries: [Bytes!]! # tuple[]
  blockNumber: BigInt!
  blockTimestamp: BigInt!
  transactionHash: Bytes!
}

type EventUpdated @entity(immutable: true) {
  id: Bytes!
  modifierAccount: Bytes! # address
  index: BigInt! # uint256
  eventUpdated_index: BigInt! # uint256
  eventUpdated_indexedEvent_ticketNumber: BigInt! # uint32
  eventUpdated_indexedEvent_reservedCapacity: BigInt! # uint32
  eventUpdated_indexedEvent_tierCount: BigInt! # uint32
  eventUpdated_indexedEvent_dateRange_start: BigInt! # uint256
  eventUpdated_indexedEvent_dateRange_end: BigInt! # uint256
  eventUpdated_indexedEvent_organizer: Bytes! # address
  eventUpdated_indexedEvent_name: Bytes! # bytes32
  eventUpdated_indexedEvent_subEvURI: String! # string
  eventUpdated_indexedEvent_eventState: Int! # uint8
  eventUpdated_indexedEvent_eventType: Int! # uint8
  eventUpdated_indexedEvent_exist: Boolean! # bool
  blockNumber: BigInt!
  blockTimestamp: BigInt!
  transactionHash: Bytes!
}

type EventsList @entity(immutable: true) {
  id: Bytes!
  indexedEvents: [Bytes!]! # tuple[]
  blockNumber: BigInt!
  blockTimestamp: BigInt!
  transactionHash: Bytes!
}

type LomEvTokensUpdate @entity(immutable: true) {
  id: Bytes!
  tokenIds: [BigInt!]! # uint256[]
  tierIdentifiers: [BigInt!]! # uint256[]
  triggerContract: Bytes! # address
  blockNumber: BigInt!
  blockTimestamp: BigInt!
  transactionHash: Bytes!
}

type MarketAllowed @entity(immutable: true) {
  id: Bytes!
  marketEntering: Bytes! # address
  date: BigInt! # uint256
  callerAccount: Bytes! # address
  blockNumber: BigInt!
  blockTimestamp: BigInt!
  transactionHash: Bytes!
}

type MetadataHashChange @entity(immutable: true) {
  id: Bytes!
  date: BigInt! # uint256
  newLomEventMetadata: String! # string
  blockNumber: BigInt!
  blockTimestamp: BigInt!
  transactionHash: Bytes!
}

type SaleExecuted @entity(immutable: true) {
  id: Bytes!
  marketPlace: Bytes! # address
  buyer: Bytes! # address
  tokenIds: [BigInt!]! # uint256[]
  blockNumber: BigInt!
  blockTimestamp: BigInt!
  transactionHash: Bytes!
}

type TierAdded @entity(immutable: true) {
  id: Bytes!
  identifer: BigInt! # uint256
  tierCount: BigInt! # uint256
  hashOfTier: Bytes! # bytes32
  blockNumber: BigInt!
  blockTimestamp: BigInt!
  transactionHash: Bytes!
}

type TierSales @entity(immutable: true) {
  id: Bytes!
  serieIdentifier: BigInt! # uint256
  tiers: [BigInt!]! # uint256[]
  tixToUpdate: [BigInt!]! # uint256[]
  blockNumber: BigInt!
  blockTimestamp: BigInt!
  transactionHash: Bytes!
}

type TierUpdated @entity(immutable: true) {
  id: Bytes!
  modifierAccount: Bytes! # address
  date: BigInt! # uint256
  tierUpdated_index: BigInt! # uint256
  tierUpdated_indexedTier_tokenId: BigInt! # uint256
  tierUpdated_indexedTier_eventId: BigInt! # uint256
  tierUpdated_indexedTier_price: BigInt! # uint256
  tierUpdated_indexedTier_approvedToSell: BigInt! # uint32
  tierUpdated_indexedTier_totalSold: BigInt! # uint32
  tierUpdated_indexedTier_ticketNumber: BigInt! # uint32
  tierUpdated_indexedTier_reservedForTier: BigInt! # uint32
  tierUpdated_indexedTier_tierState: Int! # uint8
  tierUpdated_indexedTier_tierName: String! # string
  tierUpdated_indexedTier_exist: Boolean! # bool
  blockNumber: BigInt!
  blockTimestamp: BigInt!
  transactionHash: Bytes!
}

type TiersDeleted @entity(immutable: true) {
  id: Bytes!
  modifierAccount: Bytes! # address
  tiersDeleted: [Bytes!]! # tuple[]
  blockNumber: BigInt!
  blockTimestamp: BigInt!
  transactionHash: Bytes!
}

type TiersList @entity(immutable: true) {
  id: Bytes!
  tiersList: [Bytes!]! # tuple[]
  blockNumber: BigInt!
  blockTimestamp: BigInt!
  transactionHash: Bytes!
}

type TiersToMarket @entity(immutable: true) {
  id: Bytes!
  modifierAccount: Bytes! # address
  tiersSentToMarket: [Bytes!]! # tuple[]
  blockNumber: BigInt!
  blockTimestamp: BigInt!
  transactionHash: Bytes!
}


### Additional context

The enum in question is not used on this Library contract but on a different contract which I have the intend to include in this subgraph but as the command fails Im not able to add the new source. If I run 'graph add' separately, I get "Failed to load subgraph from subgraph.yaml: Cannot read properties of undefined (reading 'value')". 

I have no way to know what parameter is the 'value' that the cli is reading from on the .yaml file.

**Comment by azf20** [comment]
hey @nicolasguasca1 will you need that enum in your subgraph? if not the simplest way to unblock would be to grab an ABI and remove references to the enum. I don't know why enums wouldn't be supported for type conversion though cc @leoyvens?

**Comment by nicolasguasca1** [comment]
Thank you for your answer @azf20 ! I was able to remove the reference for this type while creating the abi file during the execution of the init command. It actually led me to add another source. I introduced my second contract address and it throws: "✖ Failed to load subgraph from subgraph.yaml: Cannot read properties of undefined (reading 'value')".

Is this value the reference I erased from my previous contract?

Maybe not because actually before I managee to erase the type reference on the ABI and I got the error in the first place I also tried to 'graph add' and it threw this error.
